Use rpm instead of pip for installation of system-wide Python packages

tjanez opened the pull request:
This prevents conflicts between `rpm` and `pip` (e.g. rpm complaining when updating `python-virtualenv` since all files have been modified with `pip`'s system-wide installation of `virtualenv`).
Drop installation of `python-devel` since it is not really a common requirement. Projects needing to build C extensions of Python packages should ensure it is installed separately.
Drop installation of `python-pip` to prevent users/administrators from thinking they can install system-wide Python packages with `pip`.

tjanez committed:
Use rpm instead of pip for installation of system-wide Python packages

This prevents conflicts between rpm and pip (e.g. rpm complaining when
updating python-virtualenv since all files have been modified with
pip's system-wide installation of virtualenv).
Drop installation of 'python-devel' since it is not really a common
requirement. Projects needing to build C extensions of Python packages
should ensure it is installed separately.
Drop installation of 'python-pip' to prevent users/administrators from
thinking they can install system-wide Python packages with pip.

dblenkus reviewed (changes requested):
This doesn't work with Python 3.4. After installing it (`yum install python34`) and running `virtualenv --python=/bin/python3 venv` I get the error:

    Running virtualenv with interpreter /bin/python3
    Using base prefix '/usr'
    New python executable in venv/bin/python3
    Not overwriting existing python script venv/bin/python (you must use venv/bin/python3)
    Failed to import the site module
    Traceback (most recent call last):
      File "/home/vagrant/venv/lib64/python3.4/site.py", line 67, in <module>
        import os
      File "/home/vagrant/venv/lib64/python3.4/os.py", line 618, in <module>
        from _collections_abc import MutableMapping
    ImportError: No module named '_collections_abc'
    ERROR: The executable venv/bin/python3 is not functioning
    ERROR: It thinks sys.prefix is '/home/vagrant' (should be '/home/vagrant/venv')
    ERROR: virtualenv is not compatible with this system or executable

Cause is described [here](https://github.com/pypa/virtualenv/issues/463) and recommended solution is `pip install -U virtualenv` :)

**Extra:** We should probably add an option to install python3 in common role (and set default to `True` :P)

**PS**: I agree that this is better solution than the current one, but we must make it py3 compatible.

tjanez commented:
> This doesn't work with Python 3.4

Yes, I missed that.

There seems to be no obvious solution to this, but there's been increased activity in Red Hat's Bugziila regarding this issue in the last few days:
- https://bugzilla.redhat.com/show_bug.cgi?id=1263057
- https://bugzilla.redhat.com/show_bug.cgi?id=1319963
- https://bugzilla.redhat.com/show_bug.cgi?id=1411685

I hope things will get resolved soon.

In the meantime, we would need to implement a work-around for Python 3.

> Extra: We should probably add an option to install python3 in common role (and set default to True :P)

Agreed. We might wait a little bit before setting Python 3 as the default, at least until the aforementioned issue gets resolved.

> PS: I agree that this is better solution than the current one, but we must make it py3 compatible.

👍

tjanez committed:
Add support for installing Python 3 and corresponding virtualenvs

Introduce a new configuration option 'common_python3_enabled' which
indicates whether to install Python 3 and enables creating Python 3
virtual environments.

tjanez commented:
> In the meantime, we would need to implement a work-around for Python 3.

@dblenkus, I've done that. Please, take a look.
